### PR TITLE
Add electric combat chainsaw

### PR DESCRIPTION
**Pull request by Rail-Runner**
Add craftable electric combat chainsaw.

Also move existing combat chainsaw to melee.json and remove "(off)" from combat chainsaw name. "You drop your combat chainsaw" sounds better than "You drop your combat chainsaw (off)"

### CODE REVIEW

**Comment by illi-kun** [comment]
Please don't put them in ```tools.json```, ```melee.json``` fits better since their main purpose is combat. Feel free to transfer already existing combat chainsaw to melee.json as well.

**Comment by Coolthulhu** [comment]
Diamond combat chainsaw would be just too sci-fi, and soft sci-fi at that. The method of crafting diamond blades would most likely render the chainsaw unusable.

About SHEATH_SWORD: I'm certain you can't put a chainsaw in a scabbard, not even one a zweihander would fit in. Is it about just strapping it to your back or something?

As mentioned in #14609 electric chainsaws:
* Shouldn't drain power passively, but per attack
* Should drain charges much faster than gasoline chainsaws. If we wanted realism, it should drain roughly 100 times as many charges in the turn it is used to attack. Though that would make them incredibly expensive, so making it 10-20 charges per attack would be better. Later on gasoline chainsaws should be buffed by dropping their passive drain significantly
* Should be silent when not used to attack

**Comment by Darkyhard** [comment]
> The method of crafting diamond blades would most likely render the chainsaw unusable.

Why would it make the chainsaw unusable? You're just putting a diamond edge on its' chain. I don't know much about how chainsaws actually operate, but I don't think that a diamond edge on the chain would change much.

> About SHEATH_SWORD: I'm certain you can't put a chainsaw in a scabbard, not even one a zweihander would fit in.

If that combat chainsaw is redesigned to actually be more of a sword by adding a long handle on one end or something and removing unnecessary stuff, you could probably put it in a scabbard if it was specifically designed for it, but it will most likely be a scabbard that is worn on back.

> As mentioned in #14609 electric chainsaws:
   Shouldn't drain power passively, but per attack
   Should drain charges much faster than gasoline chainsaws. If we wanted realism, it should drain roughly 100 times as many charges in the turn it is used to attack. Though that would make them incredibly expensive, so making it 10-20 charges per attack would be better. Later on gasoline chainsaws should be buffed by dropping their passive drain significantly
   Should be silent when not used to attack

That would be very difficult for me to implement, though it would be good if someone did it.

**Comment by kevingranade** [comment]
If we have a mechanism for weapons consuming charges on attack, gas
chainsaws should feature low noise and fuel consumption when on, and high
noise and fuel consumption when attacking.
The sheath is right out for chainsaws, maybe you can attach a strap and
wear them though.

**Comment by kevingranade** [comment]
Diamond swords in the game are diamond coated, if you took apart a chain
and diamond coated the links, you wouldn't be able to put it back together.
That's what coolthulu is getting at I think. My problem with it is the way
chainsaws work, it would grind the diamond back off the chain pretty
quickly.
As for scabbards, the blade is the wrong size and shape, and you still have
the handle assembly making up 90% of the weight of the weapon.

**Comment by Darkyhard** [comment]
Removed everything but electric combat chainsaw.

By the way, where do I ask if something is reasonable to have before actually starting to work on it? Having your work wasted doesn't feel good.

**Comment by Mecares** [comment]
Well if it is only a realism concern you could work around that and instead of a Diamond hardenend chainsaw made with the CVD machine, you could just create a rare item to craft it something like: stone-saw-chain: "This chain with teeth out of silicon carbide is for a saw used by masons and sculptors to cut through solid stone."  and use this to craft your hardened combat chainsaw.

**Comment by Coolthulhu** [comment]
> By the way, where do I ask if something is reasonable to have before actually starting to work on it? 

Make a thread on the forums, in the drawing board. Even something as simple as "I plan to make x and y, is it OK?".

File: src/iuse.cpp [hunk 1]
```diff
@@ -3451,6 +3451,15 @@ int iuse::combatsaw_off(player *p, item *it, bool, const tripoint& )
         _("You yank the cord, but nothing happens."));
 }
 
+int iuse::elec_combatsaw_off(player *p, item *it, bool, const tripoint& )
+{
+    return toolweapon_off(p, it,
+        true,
+        !p->is_underwater(),
+        30, _("With a snarl, the electric combat chainsaw screams to life!"),
+        _("You yank the cord, but nothing happens."));
```
Comment (kevingranade) [inline src/iuse.cpp]: An electric chainsaw doesn't have a pull cord, it has a switch.

**Comment by dbackeus** [comment]
A note on where to discuss / plan ideas for new features could be added to your CONTRIBUTING.md file, which is currently mostly concerned with technical issues and licensing.

@Darkyhard out of curiosity, did you notice and look through the CONTRIBUTING.md file before starting your work on this?

**Comment by Draconomial** [comment]
Diamond chainsaw chains are real and practical. They are not diamond coated, though. The metal of the teeth have diamonds embedded into it, because they do not dull.
diamond_chain_big

**Comment by kevingranade** [comment]
Based on the location of the diamond grit, I'm assuming it's a chain
*grinder* rather than a chain *saw*, which means it would be an even worse
weapon than a regular chainsaw. Grinders are going to get clogged with
blood and visceral in short order.

Don't we have a mod for weapons like this that don't work in the real world
but sound cool?

**Comment by Draconomial** [comment]
It chews apart concrete a lot better than it does soft clay, I'm certain. That does not make it a worse weapon. It would not be great for zombies, but it would be good against giant ants and robots.

**Comment by mugling** [comment]
Similar to #15282 closed due to lack of consensus/progress. Could be resubmitted as part of a larger mod if there was sufficient interest?

**Pull request closed without merging**